test/registration_spec.rb

```diff
@@ -127,15 +127,19 @@
       }
     end
     before do
-      expect(Registration::SwMgmt).to receive(:base_product_to_register).and_return(base_product)
-    end
-
-    it "downloads available extensions" do
       remote_product = load_yaml_fixture("remote_product.yml")
-      expect(SUSE::Connect::YaST).to receive(:show_product).and_return(remote_product)
+      allow(SUSE::Connect::YaST).to receive(:show_product).and_return(remote_product)
       # no product renames defined
-      expect(Registration::SwMgmt).to receive(:update_product_renames).with({})
+      allow(Registration::SwMgmt).to receive(:update_product_renames).with({})
+    end
 
+    it "returns empty list if no base product is found" do
+      expect(Registration::SwMgmt).to receive(:base_product_to_register).and_return(nil)
```

Comment:
NP: I like more to use `context`s, a `let` for the _base_product_ and move this "expectation" to the `before` block using an `allow` instead.